neurovault/apps/statmaps/tasks.py

```diff
@@ -24,9 +29,84 @@ def generate_glassbrain_image(image_pk):
     plt.close('all')
     return png_img_path
 
+# IMAGE TRANSFORMATION ################################################################################
+
+# Save 4mm, brain masked image vector in pkl file in image folder
+@shared_task
+def save_resampled_transformation_single(pk1, resample_dim=[4, 4, 4]):
+    from neurovault.apps.statmaps.models import Image
+    from nilearn.image import resample_img
+    import neurovault
+    img = get_object_or_404(Image, pk=pk1)
+    nii_obj = nib.load(img.file.path)
+    
+    # If a standard already exists with the voxel dimension, we use that to save time
+    standard = os.path.abspath(os.path.join(neurovault.settings.BASE_DIR,
+                                                  "static","anatomical",
+                                                  "MNI152_%smm.nii.gz" % resample_dim[0]))
+    if not os.path.exists(standard):
+      standard = os.path.abspath(os.path.join(neurovault.settings.BASE_DIR,"static","anatomical","MNI152.nii.gz"))
+    standard_brain = nib.load(standard)
+    
+    # Resample both image and mask - resampling will be skipped if affines already equivalent
+    img_resamp, ref_resamp = resample_images_ref(images=[nii_obj], 
+                                                 reference=standard_brain, 
+                                                 interpolation="continuous",
+                                                 resample_dim=resample_dim)
+
+    # Mask the image, and save pickle image folder 
+    # (this is the same procedure used to produce the atlas vector that will be used for scatterplot)
+    image_vector = img_resamp[0].get_data()[ref_resamp.get_data()!=0]
+
+    pkl_resamp_name = "transform_%smm_%s.pkl" %(resample_dim[0],img.pk)
+    pkl_img_path = os.path.join(os.path.split(img.file.path)[0], pkl_resamp_name)
+    if os.path.exists(pkl_img_path):
+        os.unlink(pkl_img_path)
+    pickle.dump(image_vector,open(pkl_img_path,"wb"))
+
+    # Update the image "transform" field with the pkl_img_path
+    img.transform = pkl_img_path
+    img.save()
+
+    return pkl_img_path
+
+
+# SIMILARITY CALCULATION ##############################################################################
 
 @shared_task
-def save_voxelwise_pearson_similarity(pk1, pk2, resample_dim=[4, 4, 4]):
+def run_voxelwise_pearson_similarity(pk1):
+
+    # Make sure we have a transform for pk in question
+    from neurovault.apps.statmaps.models import Image, Comparison
+    image1 = get_object_or_404(Image, pk=pk1)
+    if image1.transform is None:
```

Comment:
ok, I have a fix for that issue! However testing with a large(r) set of images (7), we are running into the issue again (toward the end) of the pickle file with the transformation not existing.

This is because we check to see if it exists when calculating the transform, and delete:

     if os.path.exists(pkl_img_path):
        os.unlink(pkl_img_path)

Suggestion: remove this from the function entirely - If it already exists, it will be overwritten by the new one when the (atomically done) new file is created.

I just tested this, it fixed the problem. Do you see any issues with not deleting the transform?